sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncTokenPolicy.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * A policy uses a concurrent map to maintain the latest sync-tokens. When this HTTP pipeline policy is triggered, the
+ * policy will retrieve all sync-tokens without sequence number segment from the concurrent map and use it in the HTTP
+ * request. Also after received the HTTP response, update the latest sync-tokens to the map.
+ */
+public final class SyncTokenPolicy implements HttpPipelinePolicy {
+    private static final String SYNC_TOKEN = "Sync-Token";
+    private final ConcurrentHashMap<String, SyncToken> syncTokenMap = new ConcurrentHashMap<>(); // key is sync-token id
+
+    /**
+     * Add or update the sync token to a thread safe map.
+     *
+     * @param context request context
+     * @param next The next policy to invoke.
+     * @return A {@link Mono} representing the HTTP response that will arrive asynchronously.
+     */
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+
+        // Add all of sync-tokens to HTTP request header
+        context.getHttpRequest().setHeader(SYNC_TOKEN, getSyncTokenHeader());
+
+        return next.process().flatMap(httpResponse -> {
+            // Get the sync-token from HTTP response header
+            final String syncTokenValue = httpResponse.getHeaders().getValue(SYNC_TOKEN);
+
+            // Skip sync-token updates of concurrent map if no 'Sync-Token' header
+            if (syncTokenValue == null) {
+                return Mono.just(httpResponse);
+            }
+
+            // Sync-Token header could have more than one value
+            final String[] syncTokens = syncTokenValue.split(",");
+            for (final String syncTokenStr : syncTokens) {
+                final SyncToken syncToken = SyncToken.parseSyncToken(syncTokenStr);
```

Comment:
In your static method, you  return null if it's not a synctoken. you are potentially getting a null reference exception because you use it later on without checking it is not null.